compiler/repl-service/client/src/DA/Daml/LF/ReplClient.hs

```diff
@@ -113,21 +120,29 @@ withReplClient opts@Options{..} f = withTempFile $ \portFile -> do
         , concat [ ["--max-inbound-message-size", show (getMaxInboundMessageSize size)] | Just size <- [optMaxInboundMessageSize] ]
         ]
     withCreateProcess replServer { std_out = optStdout } $ \_ stdout _ ph -> do
-      port <- readPortFile ph maxRetries portFile
-      let grpcConfig = ClientConfig (Host "127.0.0.1") (Port port) [] Nothing Nothing
-      threadDelay 1000000
-      withGRPCClient grpcConfig $ \client -> do
-          replClient <- Grpc.replServiceClient client
-          f Handle
-              { hClient = replClient
-              , hOptions = opts
-              } stdout ph
-
-loadPackage :: Handle -> BS.ByteString -> IO (Either BackendError ())
-loadPackage Handle{..} package = do
+      clientBarrier <- newBarrier
+      terminateBarrier <- newBarrier
```

Comment:
I find it a bit confusing that `terminateBarrier` doesn't have anything to do with `hTerminate`. Maybe you could rename `terminateBarrier`, or add a comment?